File: Python/parser.py
def parse(data):
    value = 0
    output = []
    for cmd in data :
        if cmd=='i':
            value+=1
        elif cmd=='d':
            value-=1
        elif cmd=='s':
            value**=2
        elif cmd=='o':
            output.append(value)
    return output

File: PHP/parser.php
<?php

    function parse($data){
        $data = strtolower($data);
        $output = [];
        $value = 0;
        for($i=0;$i<strlen($data);$i++){
            $char= $data[$i];
            switch($char):
                case 'i':
                    $value += 1;
                    break;
                case 'd':
                    $value -= 1;
                    break;
                case 's':
                    $value *= $value;
                    break;
                case 'o':
                    $output[] = $value;
                    break;
                    default:
                    break;
            endswitch;
        }
        return $output;
    }